src/index.js

```diff
@@ -55,13 +57,13 @@ export class OIDCClient {
       return;
 
     try {
-      let count = 0;
-      while (this.refreshTokenLock && count < 15) {
-        await this._wait((count > 0) ? (200 * count) : undefined); //delays the next check of refreshTokenLock
-        count += 1;
+      for (let count = 0; this.refreshTokenLock && count < 15; count++) {
+        if (this.#refreshTokenPromise) break;
```

Comment:
should we also check for the promise `state`. only break if promise is `pending`.

Reply:
resolving since we set promise to `null` on `finally`. and finding status needs a polyfill